tests/click_tests/recovery.py

```diff
@@ -0,0 +1,56 @@
+from .. import buttons
+
+
+def enter_word(debug, word):
+    word = word[:4]
+    for coords in buttons.type_word(word):
+        debug.click(coords)
+    return debug.click(buttons.CONFIRM_WORD, wait=True)
+
+
+def select_number_of_words(debug, num_of_words=20):
+    # confirm recovery
+    layout = debug.wait_layout()
+    assert layout.text.startswith("Recovery mode")
+    layout = debug.click(buttons.OK, wait=True)
+
+    # select number of words
+    assert "Select number of words" in layout.text
+    layout = debug.click(buttons.OK, wait=True)
+    assert layout.text == "WordSelector"
+
+    if num_of_words != 20:
+        raise NotImplementedError
+
+    # click "20" at 2, 2
+    coords = buttons.grid34(2, 2)
+    layout = debug.click(coords, wait=True)
+    assert "Enter any share" in layout.text
+
+
+def enter_share(debug, share: str):
+    layout = debug.click(buttons.OK, wait=True)
+
+    assert layout.text == "Slip39Keyboard"
+    for word in share.split(" "):
+        layout = enter_word(debug, word)
+
+    return layout
+
+
+def enter_shares(debug, shares: list):
+    expected_text = layout = None
```

Comment:
better:
```python
layout = debug.read_layout()
expected_text = "Enter any share"
for share in shares:
    assert expected_text in layout.text
    ...
```

Reply:
For some reason I haven't realized there is no harm in reading the layout twice (because I read it at the end of `select_number_of_words` as well). Fixed.